File: nine-language-version/lessons/02-variables-types/variables_types.js
// variables_types.js

// Variable declarations (3 ways)
var oldStyle = "avoid var";     // function-scoped (old)
let mutable = "can change";     // block-scoped
const immutable = "cannot reassign";  // block-scoped, constant

// Numbers (only one number type!)
let age = 25;                   // integer value
let price = 19.99;              // floating-point value
// Both are actually 64-bit floats

// Strings
let name = "Alice";
let greeting = 'Hello';
let template = `Hello, ${name}!`;  // template literals

// Booleans
let isStudent = true;
let hasGraduated = false;

// Null and Undefined
let nothing = null;             // intentional absence
let notDefined;                 // undefined (not initialized)

// Type checking
console.log("Type of age:", typeof age);
console.log("Type of name:", typeof name);

// Weak typing - automatic conversions
console.log('"5" + 3 =', "5" + 3);       // "53" (string concatenation)
console.log('"5" - 3 =', "5" - 3);       // 2 (numeric subtraction)
console.log('true + 1 =', true + 1);     // 2 (boolean to number)

File: nine-language-version/lessons/02-variables-types/variables_types.c
// variables_types.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    // Must declare type before use

    // Integers (various sizes)
    char small = 127;              // 8-bit integer
    short medium = 32000;          // 16-bit integer
    int age = 25;                  // 32-bit integer (usually)
    long big = 2147483647L;        // 64-bit integer

    // Floating point
    float price = 19.99f;          // 32-bit float
    double precise = 3.14159265359; // 64-bit float

    // Character
    char letter = 'A';             // single character

    // Boolean (need stdbool.h)
    bool is_student = true;

    // Strings (character arrays)
    char name[] = "Alice";

    // Cannot change type!
    // age = "text";  // ERROR: incompatible types

    printf("Age: %d\n", age);
    printf("Price: %.2f\n", price);
    printf("Name: %s\n", name);
    printf("Letter: %c\n", letter);
    printf("Is student: %d\n", is_student);

    return 0;
}
